### PR TITLE
Testgrid reporting enhancements

### PR DESCRIPTION
**Pull request by emosbaugh**
1. Changed dev environment postgres labels from app=postgres to app=testgrid-postgres to avoid collisions.
2. Added failure reason to testgrid instance run reporting.
3. The cleanup routine will now report run as failed/timeout.
4. The runner script will now always report and send logs before exiting and always exit after reporting.

### CODE REVIEW

File: testgrid/tgapi/pkg/handlers/instance_finish.go [hunk 1]
```diff
@@ -13,7 +13,8 @@ import (
 )
 
 type FinishInstanceRequest struct {
-	Success bool `json:"success"`
+	Success bool   `json:"success"`
+	Failure string `json:"failure"`
```
Comment (laverya) [inline testgrid/tgapi/pkg/handlers/instance_finish.go:17]: maybe "failureType" or something?

File: testgrid/tgapi/pkg/handlers/instance_sonobouy.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func InstanceSonobuoyResults(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	if err := testinstance.SetInstanceFinishedAndSuccess(instanceID, isSuccess); err != nil {
+	if err := testinstance.SetInstanceFinishedAndSuccess(instanceID, isSuccess, "sonobuoy_results"); err != nil {
```
Comment (laverya) [inline testgrid/tgapi/pkg/handlers/instance_sonobouy.go:44]: why hardcode `sonobuoy_results` here?

File: testgrid/tgapi/pkg/handlers/instance_finish.go (hunk 1 above)
Comment (emosbaugh) [reply testgrid/tgapi/pkg/handlers/instance_finish.go:17]: ill change to FailureReason

File: testgrid/tgapi/pkg/handlers/instance_sonobouy.go (hunk 2 above)
Comment (emosbaugh) [reply testgrid/tgapi/pkg/handlers/instance_sonobouy.go:44]: cause that is the cause of the failure

File: testgrid/tgrun/pkg/runner/cleanup.go [hunk 3]
```diff
@@ -38,6 +40,17 @@ func CleanUpVMIs() error {
 
 		// cleanup VMIs that have been running for more than one hour - the in-script timeout is 30m
 		if vmi.Status.Phase == kubevirtv1.Running && time.Since(vmi.CreationTimestamp.Time).Minutes() > 60 {
+			if apiEndpoint := vmi.Annotations["testgrid.kurl.sh/apiendpoint"]; apiEndpoint != "" {
+				url := fmt.Sprintf("%s/v1/instance/%s/finish", apiEndpoint, vmi.Name)
```
Comment (laverya) [inline testgrid/tgrun/pkg/runner/cleanup.go:44]: I LIKE this, thanks for adding it!

File: testgrid/tgapi/pkg/handlers/instance_sonobouy.go (hunk 2 above)
Comment (laverya) [reply testgrid/tgapi/pkg/handlers/instance_sonobouy.go:44]: right but it won't always be a failure, will it?

File: testgrid/tgrun/pkg/runner/embed/runcmd.sh [hunk 4]
```diff
@@ -116,33 +79,12 @@ function run_install() {
...
-        elif command_exists "crictl" ; then
-            crictl ps -a
-        fi
     fi
 
-    echo "";
-    echo "running pods after completion:";
-    kubectl get pods -A
-    echo "";
+    collect_debug_info_after_kurl || :
```
Comment (laverya) [inline testgrid/tgrun/pkg/runner/embed/runcmd.sh:84]: what does `|| :` do?

**Review by laverya — APPROVED** [review]
generally lgtm,  but I think the general pattern is `|| true` not `|| :`

Reporting failure causes, reporting timeouts as failures,  and refactoring the runcmd.sh to reduce code duplication and improve resiliency 👍

File: testgrid/tgrun/pkg/runner/embed/runcmd.sh (hunk 4 above)
Comment (emosbaugh) [reply testgrid/tgrun/pkg/runner/embed/runcmd.sh:84]: || true

**Pull request merged**